lib/response.js

```diff
@@ -196,10 +196,86 @@ function buildSpeechletResponse(options) {
     return returnResult;
 }
 
-// TODO: check for ssml content in card
+/**
+ * TODO: check for ssml content in card
+ *
+ * This function creates a response object that Alexa will say to the user.
+ * This response has two properties:
+ * type - which can be SSML or PlainText
+ * speech - Which is a string that could be an SSML message or plain text.
+ *
+ * The message passed in can be one of the followings:
+ *
+ * - string, number or boolean which will be converted to the following response:
+ *  { type: 'SSML', speech: '<speak>{message value}</speak>' }
+ *
+ * - when an object is passed in that has type and speech properties, it'll just be returned.
+ *
+ * @param message The message to convert into a object that has type + speech properties.
+ * @returns {{type: string, speech: string}}
+ */
 function getSSMLResponse(message) {
-    return {
-        type: 'SSML',
-        speech: `<speak> ${message} </speak>`
-    };
+    if (message === null || message === undefined) {
+        throw new Error('The message must not be null or undefined');
+    }
+
+    var messageType = typeof(message);
+    if (messageType === "function") {
+        throw new Error('The message must not be a function');
+    }
+
+    if (messageType === "object") {
+        if (message.type && message.speech) {
+            if (isValidSpeechType(message.type)) {
```

Comment:
no else condition? (What happens if messageType is an object with an invalid `speechType`?

(Currently the error `The message must not be of type: object` is shown, which is not correct.